enhancements/olm-cmd-workflows.md

```diff
@@ -0,0 +1,164 @@
+
+- \[ \] Enhancement is `provisional`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created.
+
+## Summary
+
+This proposal provides documentation describing the possible workflows of new commands for generating operator manifests with user inputs and publishing operator bundle image for OLM to deploy the operator.
+
+We plan to improve the developer experience for operator authors using Operator SDK with OLM by:
+1. Providing an interactive subcommand to get inputs for generating CSV.
+2. Providing a single command for generating operator manifests and for creating operator bundle image.
+
+## Motivation
+
+Currently, the `operator-sdk generate csv` command is used for generating CSVs, which scaffolds the project and writes the CSV manifest on disk. This process still requires the developers to manually edit the CSV when any of the required fields are missing. Also, with the deprecation in the use of package manifests for OLM, generating CSVs and creating operator bundle image does not require two separate commands. Integrating both the functionalities and providing a single command with an interactive input for collecting the required UI metadata to generate CSV as well as for building an operator bundle image will result in a better developer experience.
+
+## Goals
+
+The proposal is aimed at explaining the resulting process of developing an OLM enabled operator from scratch with the above mentioned improvements.
+
+## Proposal
+
+### README
+
+The following document discusses workflows for creating a new operator and running it with OLM:
+
+Operator creation workflow:
+
+1. Create a new operator project using the SDK Command Line Interface(CLI)
+2. Define new resource APIs by adding Custom Resource Definitions(CRD)
+3. Define Controllers to watch and reconcile resources
+4. Write the reconciling logic for your Controller using the SDK and controller-runtime APIs
+5. Use the SDK CLI to build and generate the operator deployment manifests
+
+OLM Workflow:
+
+1. Use `operator-sdk generate bundle` to generate kustomize base templates with CSV metadata, build and push the operator bundle image to the specified registry.
+  * If the kustomize template having the operator and UI metadata is present, it is utilized along with the existing CRDs to build the operator image bundle.
+  * If the kustomize template is not present, provide inputs regarding the UI metadata to the interactive prompts appearing further.
+2. Use `operator-sdk run --olm` to generate kustomize base template with operator metadata, build and run the operator.
+
+Command details:
+
+1. `operator-sdk generate bundle`:
+
+The `operator-sdk generate bundle` command will create a kustomize base template containing metadata for the generation of CSV, build and push the operator image bundle to the specified registry.
```

Comment:
Do we want `generate bundle` to be concerned with building and pushing images?

I recall there being discussion that `generate bundle` would create/update the ./bundle ensure a Dockerfile exists (perhaps called `./bundle.Dockerfile`), and then the user can use whatever tooling they want to build and push.

Reply:
Updated the command workflow for `generate bundle` to create customize base template and dockerfile.